pkg/platform/aws/template.go

```diff
@@ -100,6 +97,9 @@ module "aws-{{.Config.ClusterName}}" {
 
   {{- if .Config.EncryptPodTraffic }}
   encrypt_pod_traffic = {{.Config.EncryptPodTraffic}}
+  network_mtu = {{ .Config.NetworkMTU }} - 40 # Wireguard overhead
+  {{- else }}
+  network_mtu = {{ .Config.NetworkMTU }} - 20 # IP in IP overhead
```

Comment:
I think this change of meaning of `network_mtu` parameter should be done in separate commit, before we add encryption support.

Also, I wonder at which point we should reduce the MTU. Maybe it should be done on the Helm template level, so the mutation is closer to the point where it's used?